app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -348,11 +348,21 @@ def import_enabled_repositories(repos)
       end
 
       def import_module_streams(module_streams)
+        # create_or_find_by avoids race conditions during concurrent registrations but clogs postgres logs with harmless errors.
```

Comment:
instead of two different implementation, would it be possible to catch the "postgres logs with harmless errors" and handle this accordingly?

Reply:
No, that's the way the race condition is avoided - by letting the Postgres constraint handle it rather than ruby/rails.

Reply:
ok. make sense.